dpnp/tests/third_party/cupy/core_tests/test_ndarray_adv_indexing.py

```diff
@@ -0,0 +1,947 @@
+        "value": 1,
+    },
+    {"shape": (2, 3, 4), "indexes": (slice(None), 2, False), "value": 1},
+    {
+        "shape": (2, 3, 4),
+        "indexes": (
+            slice(None),
+            slice(None),
+            numpy.random.choice([False, True], (4,)),
+        ),
+        "value": 1,
+    },
+    {
+        "shape": (2, 3, 4),
+        "indexes": (numpy.random.choice([False, True], (2, 3)),),
+        "value": 1,
+    },
+    {
+        "shape": (2, 3, 4),
+        "indexes": (
+            slice(None),
+            numpy.random.choice([False, True], (3, 4)),
+        ),
+        "value": 1,
+    },
+    {
+        "shape": (2, 3, 4),
+        "indexes": (numpy.random.choice([False, True], (2, 3, 4)),),
+        "value": 1,
+    },
+    # {'shape': (2, 3, 4),
+    #  'indexes': (1, None, numpy.array([True, False, True])), 'value': 1},  dpctl-1911
+    # multiple arrays
+    {"shape": (2, 3, 4), "indexes": ([0, -1], [1, -1]), "value": 1},
+    {"shape": (2, 3, 4), "indexes": ([0, -1], [1, -1], [2, 1]), "value": 1},
+    {"shape": (2, 3, 4), "indexes": ([0, -1], 1), "value": 1},
+    # {'shape': (2, 3, 4), 'indexes': ([0, -1], slice(None), [1, -1]),
+    #  'value': 1},  #dpctl-1912
+    {"shape": (2, 3, 4), "indexes": ([0, -1], 1, 2), "value": 1},
+    # {'shape': (2, 3, 4), 'indexes': ([1, 0], slice(None), [[2, 0], [3, 1]]),
+    #  'value': 1},  #dpctl-1912
+    # multiple arrays and basic indexing
+    # {'shape': (2, 3, 4), 'indexes': ([0, -1], None, [1, 0]), 'value': 1},  #dpctl-1912
+    # {'shape': (2, 3, 4), 'indexes': ([0, -1], slice(0, 2), [1, 0]),
+    #  'value': 1},  #dpctl-1912
+    # {'shape': (2, 3, 4), 'indexes': ([0, -1], None, slice(0, 2), [1, 0]),
+    #  'value': 1},  #dpctl-1912
+    {
+        "shape": (1, 1, 2, 3, 4),
+        "indexes": (None, slice(None), slice(None), [1, 0], [2, -1], 1),
+        "value": 1,
+    },
+    # {'shape': (1, 1, 2, 3, 4),
+    #  'indexes': (None, slice(None), 0, [1, 0], slice(0, 2, 2), [2, -1]),
+    #  'value': 1},  #dpctl-1912
+    {
+        "shape": (2, 3, 4),
+        "indexes": (slice(None), [0, -1], [[1, 0], [0, 1], [-1, 1]]),
+        "value": 1,
+    },
+    # empty arrays
+    {"shape": (2, 3, 4), "indexes": [], "value": 1},
+    {"shape": (2, 3, 4), "indexes": [], "value": numpy.array([1, 1, 1, 1])},
+    {
+        "shape": (2, 3, 4),
+        "indexes": [],
+        "value": numpy.random.uniform(size=(3, 4)),
+    },
+    {
+        "shape": (2, 3, 4),
+        "indexes": numpy.array([], dtype=numpy.int32),
+        "value": 1,
+    },
+    {
+        "shape": (2, 3, 4),
+        "indexes": numpy.array([[]], dtype=numpy.int32),
+        "value": numpy.random.uniform(size=(3, 4)),
+    },
+    {"shape": (2, 3, 4), "indexes": (slice(None), []), "value": 1},
+    {"shape": (2, 3, 4), "indexes": ([], []), "value": 1},
+    # {'shape': (2, 3, 4), 'indexes': numpy.array([], dtype=numpy.bool_),
+    #  'value': 1},  #dpctl-1913
```

Comment:
Could we include cases that are relate to dpctl-1913 since it is resolved now?

Reply:
The fix is not yet available in the internal CI. I'd propose to keep that commented until that.